### PR TITLE
fix(cmake): Make some dependency scopes private

### PR DESCRIPTION
**Pull request by geraldcombs**
Make the base sinsp dependency scopes PRIVATE. This fixes compilation here on macOS when BUILD_SHARED_LIBS=ON.

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

/area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/CMakeLists.txt [hunk 1]
```diff
@@ -309,8 +309,8 @@ if(NOT WIN32)
 			target_link_libraries(sinsp INTERFACE "${ONIGURUMA_LIB}")
 			list(APPEND SINSP_PKGCONFIG_LIBRARIES "${ONIGURUMA_LIB}")
 			add_dependencies(sinsp jq)
-		endif()
-	endif() # NOT APPLE
+		endif() # NOT APPLE
+	endif() # NOT WIN32
```
Comment (LucaGuerra) [inline userspace/libsinsp/CMakeLists.txt:313]: It still looks to me that this is `NOT APPLE`

Comment (geraldcombs) [reply userspace/libsinsp/CMakeLists.txt]: Fixed. Thanks!

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Pull request merged**